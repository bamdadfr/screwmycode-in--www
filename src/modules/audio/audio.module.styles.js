import styled from 'styled-components';

export const Invisible = styled.div`
  display: none;
`;
