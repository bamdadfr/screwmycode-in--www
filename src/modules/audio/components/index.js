export { NativeComponent } from './native'

export { HowlerComponent } from './howler'