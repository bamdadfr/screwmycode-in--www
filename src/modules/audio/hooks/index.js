export { useAutoplay } from './use-autoplay'

export { useCache } from './use-cache'

export { useInterval } from '../../../hooks/use-interval'

export { useAudioModule } from './use-audio-module'