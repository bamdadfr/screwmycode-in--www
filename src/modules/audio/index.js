export { AudioModule } from './audio.module'