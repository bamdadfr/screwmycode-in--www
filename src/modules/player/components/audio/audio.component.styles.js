import styled from 'styled-components'

export const Container = styled.div`
    height: 3em;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;

    & > audio {
        width: 100%;
    }
`