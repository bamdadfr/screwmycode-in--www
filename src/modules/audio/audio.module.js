import React from 'react'
import PropTypes from 'prop-types'
import { isFirefox } from 'react-device-detect'
import { NativeComponent, HowlerComponent } from './components'
import { useAudioModule } from './hooks'
import { Invisible } from './audio.module.styles'

const propTypes = {
    'url': PropTypes.string.isRequired,
}

/**
 * @typedef {string} Url
 * @param {{Url}} props react props
 * @returns {React.ReactElement} react component
 */
export function AudioModule ({ url }) {

    useAudioModule ()

    return (
        <>
            <Invisible>
                {isFirefox
                    ? <NativeComponent url={url}/>
                    : <HowlerComponent url={url}/>
                }
            </Invisible>
        </>
    )

}

AudioModule.propTypes = propTypes
