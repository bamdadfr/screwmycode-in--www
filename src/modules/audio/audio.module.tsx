import React, {ReactElement} from 'react';
import {NativeComponent} from './components/native/native.component';
import {useAudioModule} from './hooks/use-audio-module';
import {Invisible} from './audio.module.styles';

interface AudioModuleProps {
  url: string;
}

/**
 * Component to play audio files
 */
export function AudioModule({url}: AudioModuleProps): ReactElement {
  useAudioModule();

  return (
    <>
      <Invisible>
        <NativeComponent url={url} />
      </Invisible>
    </>
  );
}
