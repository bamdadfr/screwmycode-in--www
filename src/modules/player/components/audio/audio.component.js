import React from 'react'
import PropTypes from 'prop-types'
import { useAudioComponent } from './hooks'

const propTypes = {
    'url': PropTypes.string.isRequired,
}

/**
 * @param {object} props props
 * @param {string} props.url url
 * @returns {React.ReactElement} react component
 */
export function AudioComponent ({ url }) {

    const { ref, autoplay } = useAudioComponent (url)

    return (
        <>
            <audio
                ref={ref}
                aria-label="player"
                controls
                autoPlay={autoplay}
            >
                <track kind="captions"/>
            </audio>
        </>
    )

}

AudioComponent.propTypes = propTypes