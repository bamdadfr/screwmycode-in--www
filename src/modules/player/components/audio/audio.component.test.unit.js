import React from 'react'
import { AudioComponent } from './audio.component'
import { JestInstance } from '../../../../../jest/jest-instance'

const url = 'https://dummy.audio.url'

const instance = JestInstance (
    <AudioComponent
        url={url}
        playbackRate={1}
        loop
        volume={1}
        handleVolume={jest.fn ()}
        autoplay={false}
    />,
)

it ('should mount and match', () => {

    expect (instance).toBeDefined ()

    expect (instance).toMatchSnapshot ()

})

it ('should have a classname prop from styled components', () => {

    expect (instance.props.className).toBeDefined ()

    expect (instance.props.className).toContain ('sc')

})

it ('should have 1 children', () => {

    expect (instance.children).toHaveLength (1)

})
