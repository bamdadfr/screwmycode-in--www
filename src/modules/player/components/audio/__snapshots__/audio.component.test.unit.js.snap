// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should mount and match 1`] = `
<div
  className="sc-o9n5et-0 jVNRjU"
>
  <audio
    aria-label="player"
    autoPlay={false}
    controls={true}
  >
    <track
      kind="captions"
    />
  </audio>
</div>
`;
